doc/reference/algorithms/shortest_paths/dijkstra.rst

```diff
@@ -0,0 +1,115 @@
+Dijkstra's Algorithm
+===================
+
+Dijkstra's algorithm is a classical algorithm for finding the shortest paths
+from a single source node to all other nodes in a weighted graph with
+non-negative edge weights. It was conceived by Edsger W. Dijkstra in 1956
+and is widely used in routing, network optimization, and pathfinding problems.
+
+For a general overview in the shortest path problem see
+`Shortest Paths <index.html>`_.
+
+Problem Definition
+------------------
+Given a weighted graph G = (V, E) and a source vertex s ∈ V, compute the
+shortest-path distances d(s, v) from s to every vertex v ∈ V, where each
+edge (u, v) ∈ E has a non-negative weight w(u, v). Optionally, the algorithm
+can also produce the actual shortest paths.
+
+Algorithm
+---------
+Dijkstra's algorithm is a greedy, iterative algorithm. The main idea is to
+incrementally build a set of nodes with known shortest distances,
+selecting at each step the node with the smallest tentative distance.
+
+**Initialization**
+
+1. Assign every node a tentative distance value:
+    * `dist[s] = 0` for the source
+    * `dist[v] = ∞` for all other nodes
+2. Initialize all nodes as unvisited.
```

Comment:
Similarly, do we need to mark them as unvisited? Or just not mark them as visited? Why do we include this step in the description? People trying to understand it maybe don't need to pay attention to this.